File: main/acm-icpc-team/acm_icpc_team.py
#!/usr/bin/env python3

best = -1       # the most topics any one team knows
best_teams = 0  # count of teams that know that many

n, _ = map(int, input().split()) # don't need m
a = [int(input(), 2) for _ in range(n)]

for i in range(0, n - 1):
    for j in range(i + 1, n):
        cur = bin(a[i] | a[j]).count('1')
        
        if best < cur:
            best = cur
            best_teams = 1
        elif best == cur:
            best_teams += 1

print(best)
print(best_teams)

File: main/acm-icpc-team/acm-icpc-team.pl
#!/usr/bin/env perl

use strict;
use warnings;

my $best = -1;          # the most topics any one team knows
my $best_teams = 0;     # count of teams that know that many

<>; # don't need n, m
chomp(my @a = <>);

for my $i (0 .. $#a - 1) {
    for my $j ($i + 1 .. $#a) {
        my $cur = ($a[$i] | $a[$j]) =~ tr/1//;

        if ($best < $cur) {
            $best = $cur;
            $best_teams = 1;
        } elsif ($best == $cur) {
            ++$best_teams;
        }
    }
}

print "$best\n$best_teams\n";
